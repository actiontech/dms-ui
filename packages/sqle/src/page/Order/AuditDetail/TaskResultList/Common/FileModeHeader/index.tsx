import { Space } from 'antd';
import { IconEllipse } from '@actiontech/shared/lib/Icon/common';
import { useTranslation } from 'react-i18next';

const FileModeHeader: React.FC = () => {
  const { t } = useTranslation();
  return (
    <Space className="file-mode-title">
      <IconEllipse />

      <span className="file-mode-title-text">
        {t('audit.fileModeExecute.headerTitle')}
      </span>
    </Space>
  );
};

export default FileModeHeader;
