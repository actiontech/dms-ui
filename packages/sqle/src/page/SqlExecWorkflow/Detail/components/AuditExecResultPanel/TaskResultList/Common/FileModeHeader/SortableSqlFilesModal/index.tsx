import { BasicButton, BasicModal, BasicTable } from '@actiontech/dms-kit';
import { Alert, Space, message } from 'antd';
import { useTranslation } from 'react-i18next';
import {
  SortableSQLFilesModalProps,
  SortableSQLFilesTableRowProps
} from './index.type';
import { SortableSqlFilesModalColumns } from './columns';
import type { DragEndEvent } from '@dnd-kit/core';
import {
  DndContext,
  PointerSensor,
  useSensor,
  useSensors
} from '@dnd-kit/core';
import { restrictToVerticalAxis } from '@dnd-kit/modifiers';
import {
  arrayMove,
  SortableContext,
  useSortable,
  verticalListSortingStrategy
} from '@dnd-kit/sortable';
import { useRef, useState } from 'react';
import { CSS } from '@dnd-kit/utilities';
import { useRequest } from 'ahooks';
import {
  IAuditFileStatistic,
  IFileToSort
} from '@actiontech/shared/lib/api/sqle/service/common';
import task from '@actiontech/shared/lib/api/sqle/service/task';
import InfiniteScroll from 'react-infinite-scroll-component';
import { useTableRequestError } from '@actiontech/shared/lib/components/ActiontechTable';
import { ResponseCode } from '@actiontech/dms-kit';
import { useCurrentProject } from '@actiontech/shared/lib/features';
const SortableSqlFilesModal: React.FC<SortableSQLFilesModalProps> = ({
  open,
  onClose,
  taskId,
  refresh,
  workflowId
}) => {
  const { t } = useTranslation();
  const [messageApi, messageContextHolder] = message.useMessage();
  const { projectName } = useCurrentProject();
  const draggedItemCollection = useRef<Set<string>>(new Set());
  const submit = () => {
    const filesToSort: IFileToSort[] = Array.from(
      draggedItemCollection.current
    ).map((v) => {
      return {
        file_id: Number(v),
        new_index: (dataSource.findIndex((item) => item.file_id === v) ?? 0) + 1
      };
    });
    task
      .updateSqlFileOrderV1({
        files_to_sort: filesToSort,
        workflow_id: workflowId ?? '',
        task_id: taskId ?? '',
        project_name: projectName
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          closeModal();
          refresh();
          messageApi.success(
            t(
              'execWorkflow.audit.fileModeExecute.sortableSQLFilesModal.successTips'
            )
          );
        }
      });
  };
  const [dataSource, setDataSource] = useState<IAuditFileStatistic[]>([]);
  const { handleTableRequestError, requestErrorMessage } =
    useTableRequestError();
  const {
    run: fetchMoreData,
    data,
    loading
  } = useRequest(
    () => {
      const page = dataSource ? Math.ceil(dataSource.length / 20) + 1 : 1;
      return handleTableRequestError(
        task.getAuditFileList({
          task_id: taskId,
          page_index: `${page}`,
          page_size: '20'
        })
      ).then((res) => {
        setDataSource([...dataSource, ...(res.list ?? [])]);
        return res;
      });
    },
    {
      ready: open
    }
  );
  const resetDataSource = () => {
    handleTableRequestError(
      task.getAuditFileList({
        task_id: taskId,
        page_index: '1',
        page_size: '20'
      })
    ).then((res) => {
      setDataSource(res.list ?? []);
      draggedItemCollection.current = new Set();
      return res;
    });
  };
  const closeModal = () => {
    onClose();
    setDataSource([]);
    draggedItemCollection.current = new Set();
  };

  //drag
  const sensors = useSensors(
    useSensor(PointerSensor, {
      activationConstraint: {
        // https://docs.dndkit.com/api-documentation/sensors/pointer#activation-constraints
        distance: 1
      }
    })
  );
  const onDragEnd = ({ active, over }: DragEndEvent) => {
    if (active.id !== over?.id) {
      setDataSource((prev) => {
        const activeIndex = prev.findIndex((i) => i.file_id === active.id);
        const overIndex = prev.findIndex((i) => i.file_id === over?.id);
        return arrayMove(prev, activeIndex, overIndex);
      });

      // 前端只负责记录哪些列被拖拽过，最终提交这些拖拽过的项以及最终 index，后端处理变更
      draggedItemCollection.current.add(active.id as string);
    }
  };
  const TableRow = (props: SortableSQLFilesTableRowProps) => {
    const {
      attributes,
      listeners,
      setNodeRef,
      transform,
      transition,
      isDragging
    } = useSortable({
      id: props['data-row-key']
    });
    const style: React.CSSProperties = {
      ...props.style,
      transform: CSS.Translate.toString(transform),
      transition,
      cursor: 'move',
      ...(isDragging
        ? {
            position: 'relative',
            zIndex: 9999
          }
        : {})
    };
    return (
      <tr
        {...props}
        ref={setNodeRef}
        style={style}
        {...attributes}
        {...listeners}
      />
    );
  };
  return (
    <>
      {messageContextHolder}
      <BasicModal
        size="large"
        onCancel={closeModal}
        open={open}
        title={t(
          'execWorkflow.audit.fileModeExecute.sortableSQLFilesModal.title'
        )}
        footer={
          <Space>
            <BasicButton onClick={closeModal}>{t('common.close')}</BasicButton>
            <BasicButton onClick={resetDataSource}>
              {t(
                'execWorkflow.audit.fileModeExecute.sortableSQLFilesModal.resetFileOrder'
              )}
            </BasicButton>
            <BasicButton onClick={submit} type="primary">
              {t('common.submit')}
            </BasicButton>
          </Space>
        }
      >
        <Alert
          style={{
            marginBottom: 12
          }}
          message={t(
            'execWorkflow.audit.fileModeExecute.sortableSQLFilesModal.tips'
          )}
          type="info"
        />

        <DndContext
          sensors={sensors}
          modifiers={[restrictToVerticalAxis]}
          onDragEnd={onDragEnd}
        >
          <SortableContext
            items={dataSource.map((i) => i.file_id!)}
            strategy={verticalListSortingStrategy}
          >
            <InfiniteScroll
              hasMore={dataSource.length < (data?.total ?? 0)}
              next={fetchMoreData}
              loader={null}
              dataLength={dataSource?.length || 0}
              scrollableTarget="scrollable-target"
            >
              <div
                id="scrollable-target"
                style={{
                  maxHeight: '600px',
                  overflow: 'auto'
                }}
              >
                <BasicTable
                  loading={loading}
                  errorMessage={requestErrorMessage}
                  rowKey="file_id"
                  className="clear-padding-bottom"
                  components={{
                    body: {
                      row: TableRow
                    }
                  }}
                  dataSource={dataSource.map((v, index) => ({
                    ...v,
                    index: index + 1
                  }))}
                  columns={SortableSqlFilesModalColumns()}
                  pagination={false}
                />
              </div>
            </InfiniteScroll>
          </SortableContext>
        </DndContext>
      </BasicModal>
    </>
  );
};
export default SortableSqlFilesModal;
