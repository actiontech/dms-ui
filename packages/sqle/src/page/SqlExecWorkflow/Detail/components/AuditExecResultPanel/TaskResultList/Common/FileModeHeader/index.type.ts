import { WorkflowRecordResV2StatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export type FileModeHeaderProps = {
  taskId: string;
  refresh: () => void;
  workflowStatus?: WorkflowRecordResV2StatusEnum;
};
