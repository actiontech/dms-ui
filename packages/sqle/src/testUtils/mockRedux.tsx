import whitelist from '../store/whitelist';
import React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import sqlManagement from '../store/sqlManagement';
import sqlManagementException from '../store/sqlManagementException';
import pipeline from '../store/pipeline';

const reducers = combineReducers({
  whitelist,
  sqlManagement,
  sqlManagementException,
  pipeline
});

export const storeFactory = (initStore: Dictionary = {}) => {
  return configureStore({
    reducer: reducers,
    preloadedState: initStore
  });
};

export const CustomProvider: React.FC<{
  initStore: Dictionary;
  children: JSX.Element;
}> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
