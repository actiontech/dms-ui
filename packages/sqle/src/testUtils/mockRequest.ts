import { AxiosResponse } from 'axios';
import ruleTemplate from '@actiontech/shared/lib/api/sqle/service/rule_template';
import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import audit_plan from '@actiontech/shared/lib/api/sqle/service/audit_plan';
import { driverMeta } from '../hooks/useDatabaseType/index.test.data';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData
  };
};

export const failedData = (data?: any, otherData?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
    ...otherData
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data)
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data)
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {}
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData)
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {}
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data, otherData)
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data)
      });
    }, 3000);
  });
};

export const mockDriver = () => {
  const spy = jest.spyOn(configuration, 'getDriversV2');
  spy.mockImplementation(() => resolveThreeSecond(driverMeta));
  return spy;
};

export const AuditPlanTypesData = [
  { type: 'default', desc: '自定义', instance_type: '' },
  { type: 'mysql_slow_log', desc: '慢日志', instance_type: 'MySQL' },
  { type: 'mysql_mybatis', desc: 'Mybatis 扫描', instance_type: 'MySQL' },
  { type: 'mysql_schema_meta', desc: '库表元数据', instance_type: 'MySQL' },
  {
    type: 'ali_rds_mysql_slow_log',
    desc: '阿里RDS MySQL慢日志',
    instance_type: 'MySQL'
  },
  {
    type: 'ali_rds_mysql_audit_log',
    desc: '阿里RDS MySQL审计日志',
    instance_type: 'MySQL'
  },
  { type: 'oracle_top_sql', desc: 'Oracle TOP SQL', instance_type: 'Oracle' },
  { type: 'all_app_extract', desc: '应用程序SQL抓取', instance_type: '' },
  { type: 'tidb_audit_log', desc: 'TiDB审计日志', instance_type: 'TiDB' },
  {
    type: 'ocean_base_for_mysql_mybatis',
    desc: 'Mybatis 扫描',
    instance_type: ''
  },
  { type: 'ocean_base_for_mysql_top_sql', desc: 'Top SQL', instance_type: '' }
];
export const mockUseAuditPlanTypes = () => {
  const spy = jest.spyOn(audit_plan, 'getAuditPlanTypesV1');
  spy.mockImplementation(() => resolveThreeSecond(AuditPlanTypesData));
  return spy;
};

export const mockUseRuleType = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTypeByDBTypeV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        rule_count: 0,
        rule_type: 'DML1'
      },
      {
        rule_count: 3,
        rule_type: 'DML2'
      }
    ])
  );

  return spy;
};
