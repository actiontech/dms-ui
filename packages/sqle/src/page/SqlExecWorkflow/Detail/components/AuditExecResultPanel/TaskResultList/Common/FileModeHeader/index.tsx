import { useTranslation } from 'react-i18next';
import { BasicButton, EmptyBox } from '@actiontech/dms-kit';
import { useTypedParams } from '@actiontech/shared';
import SortableSqlFilesModal from './SortableSqlFilesModal';
import { useBoolean } from 'ahooks';
import { Typography } from 'antd';
import { FileModeHeaderProps } from './index.type';
import { RingPieFilled } from '@actiontech/icons';
import { ROUTE_PATHS } from '@actiontech/dms-kit';
const FileModeHeader: React.FC<FileModeHeaderProps> = ({
  taskId,
  refresh,
  allowExec
}) => {
  const { t } = useTranslation();
  const { workflowId } =
    useTypedParams<typeof ROUTE_PATHS.SQLE.SQL_EXEC_WORKFLOW.detail>();
  const [
    sortableSQLFilesModalOpen,
    { setFalse: closeSortableSQLFilesModal, setTrue: openSortableSQLFilesModal }
  ] = useBoolean(false);
  return (
    <div className="file-mode-title flex-space-between full-width-element">
      <div className="flex-display flex-align-center">
        <RingPieFilled className="custom-icon-ellipse" />
        <Typography.Text className="file-mode-title-text">
          {t('execWorkflow.audit.fileModeExecute.headerTitle')}
        </Typography.Text>
      </div>

      {/* #if [ee] */}
      <EmptyBox if={allowExec}>
        <BasicButton onClick={openSortableSQLFilesModal} type="primary">
          {t('execWorkflow.audit.fileModeExecute.extraButtonText')}
        </BasicButton>

        <SortableSqlFilesModal
          workflowId={workflowId ?? ''}
          refresh={refresh}
          open={sortableSQLFilesModalOpen}
          onClose={closeSortableSQLFilesModal}
          taskId={taskId}
        />
      </EmptyBox>
      {/* #endif */}
    </div>
  );
};
export default FileModeHeader;
