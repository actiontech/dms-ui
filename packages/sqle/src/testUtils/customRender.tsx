import {
  RenderResult,
  render,
  renderHook as renderHookByReact
} from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { storeFactory } from './mockRedux';
import { ConfigProvider, theme as antdTheme } from 'antd';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';

import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState)
    }
  } as any);
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHookByReact(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <StyledEngineProvider injectFirst>
              <ThemeProvider theme={themeData}>{children}</ThemeProvider>
            </StyledEngineProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: themeData
    }
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
