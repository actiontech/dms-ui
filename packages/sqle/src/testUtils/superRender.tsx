import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
