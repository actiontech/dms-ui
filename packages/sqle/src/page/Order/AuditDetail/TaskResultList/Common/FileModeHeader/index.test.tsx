import FileModeHeader from '.';
import { superRender } from '../../../../../../testUtils/customRender';

describe('test TaskResultList/FileModeHeader', () => {
  it('should match snapshot', () => {
    expect(superRender(<FileModeHeader />).container).toMatchSnapshot();
  });
});
