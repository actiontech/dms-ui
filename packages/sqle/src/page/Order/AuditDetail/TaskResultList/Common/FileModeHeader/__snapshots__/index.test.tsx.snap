// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TaskResultList/FileModeHeader should match snapshot 1`] = `
<div>
  <div
    class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center file-mode-title"
  >
    <div
      class="ant-space-item"
      style="margin-right: 8px;"
    >
      <span
        class="custom-icon-ellipse css-u3jcp4"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#E8F3FF"
            height="16"
            rx="3"
            width="16"
          />
          <circle
            cx="8"
            cy="8"
            fill="white"
            r="3"
            stroke="#4583FF"
            stroke-width="2"
          />
        </svg>
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <span
        class="file-mode-title-text"
      >
        文件信息概览
      </span>
    </div>
  </div>
</div>
`;
